arch/arm/src/common/Make.defs

```diff
@@ -0,0 +1,83 @@
+############################################################################
+# arch/arm/src/common/Make.defs
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.  The
+# ASF licenses this file to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+############################################################################
+
+# Common ARM files
+
+CMN_CSRCS += arm_allocateheap.c arm_assert.c arm_blocktask.c
+CMN_CSRCS += arm_createstack.c arm_exit.c arm_fullcontextrestore.c
+CMN_CSRCS += arm_initialize.c arm_interruptcontext.c arm_lowputs.c
+CMN_CSRCS += arm_mdelay.c arm_modifyreg16.c arm_modifyreg32.c
+CMN_CSRCS += arm_modifyreg8.c arm_puts.c arm_releasepending.c
+CMN_CSRCS += arm_releasestack.c arm_reprioritizertr.c arm_saveusercontext.c
+CMN_CSRCS += arm_stackframe.c arm_switchcontext.c arm_udelay.c
+CMN_CSRCS += arm_vfork.c arm_unblocktask.c arm_usestack.c
+
+ifeq ($(CONFIG_STACK_COLORATION),y)
+  CMN_CSRCS += arm_checkstack.c
+endif
+
+ifneq ($(CONFIG_ARCH_IDLE_CUSTOM),y)
+  CMN_CSRCS += arm_idle.c
+endif
+
+ifeq ($(CONFIG_BUILD_PROTECTED),y)
+CMN_CSRCS += arm_task_start.c arm_pthread_start.c
+CMN_CSRCS += arm_signal_dispatch.c
+CMN_UASRCS += arm_signal_handler.S
+endif
+
+ifeq ($(CONFIG_ARM_SEMIHOSTING_SYSLOG),y)
+  CMN_CSRCS += arm_semi_syslog.c
+endif
+
+ifeq ($(CONFIG_ARM_SEMIHOSTING_HOSTFS),y)
+  CMN_CSRCS += arm_hostfs.c
+endif
+
+ifeq ($(CONFIG_SCHED_THREAD_LOCAL),y)
+  CMN_CSRCS += arm_tls.c
+endif
+
+ifeq ($(CONFIG_SCHED_BACKTRACE),y)
+  ifeq ($(CONFIG_ARM_THUMB),)
+    ifeq ($(CONFIG_FRAME_POINTER),y)
+      CMN_CSRCS += arm_backtrace_fp.c
+    else
+      CMN_CSRCS += arm_backtrace_thumb.c
```

Comment:
so we still use `arm_backtrace_thumb.c` even if `CONFIG_ARM_THUMB=n`?

Reply:
Oops, should be
still use `arm_backtrace_thumb.c` even if `CONFIG_ARM_THUMB=y`!

Reply:
should we change to:
```
ifeq ($(CONFIG_FRAME_POINTER),y)
  CMN_CSRCS += arm_backtrace_fp.c
else ifeq ($(CONFIG_ARM_THUMB),y)
  CMN_CSRCS += arm_backtrace_thumb.c
endif
```

Reply:
> should we change to:
> 
> ```
> ifeq ($(CONFIG_FRAME_POINTER),y)
>   CMN_CSRCS += arm_backtrace_fp.c
> else ifeq ($(CONFIG_ARM_THUMB),y)
>   CMN_CSRCS += arm_backtrace_thumb.c
> endif
> ```

As long as ARM_THUMB is enabled, we can not use the frame pointer implement

```
          CONFIG_ARM_THUMB         CONFIG_FRAME_POINTER     CONFIG_ARM_THUMB && CONFIG_FRAME_POINTER
armv*-a/r arm_backtrace_thumb.c    arm_backtrace_fp.c       arm_backtrace_thumb.c
armv*-m   arm_backtrace_thumb.c    arm_backtrace_thumb.c    arm_backtrace_thumb.c
```

Reply:
It depend on toochain, CONFIG_FRAME_POINTER can work with clang even with CONFIG_ARM_THUMB

Reply:
ok, I added the toolchain check, please review again